api/v1alpha1/etcdpeer_types.go

```diff
@@ -53,12 +59,68 @@ type EtcdPeerSpec struct {
 	// instructions if it already knows where it's peers are.
 	// +optional
 	Bootstrap *Bootstrap `json:"bootstrap,omitempty"`
+
+	// Storage is the configuration of the disks and mount points of the Etcd
+	// pod.
+	Storage *EtcdPeerStorage `json:"storage,omitempty"`
+}
+
+// EtcdPeerStorage defines the desired storage for an EtcdPeer
+type EtcdPeerStorage struct {
+	// VolumeClaimTemplates is a claim that pods are allowed to reference.
+	// The EtcdPeer controller will create a new PersistentVolumeClaim using the
+	// StorageClass and the Storage Resource Request in this template.
+	// That PVC will then be mounted in the Pod for this EtcdPeer and the Etcd
+	// process when it starts will persist its data to the PV bound to that PVC.
+	VolumeClaimTemplate *corev1.PersistentVolumeClaimSpec `json:"volumeClaimTemplate,omitempty"`
+}
+
+func validatePersistentVolumeClaimSpec(path *field.Path, o *corev1.PersistentVolumeClaimSpec) field.ErrorList {
```

Comment:
IMO validation and defaulting code should not exist in this package. Not really critical right now, but this maybe should move out in future I think (even if it's just a sub-package of this one).

Reply:
Moved to a separate go file at least. Done.